src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,473 @@
+    )
+    parser.add_argument(
+        "--cache_dir",
+        type=Path,
+        required=False,
+        default="$NXF_SINGULARITY_CACHEDIR",
+        metavar="SINGULARITY_CACHE",
+        help="Custom path to user generated singularity container housing ncbi tool 'datasets'",
+    )
+    parser.add_argument(
+        "--datasets_batch_size",
+        type=int,
+        required=False,
+        default=100,
+        metavar="BATCH_SIZE",
+        help="Number of accessions requested in one query to datasets",
+    )
+
+    parser.add_log_arguments(add_log_file=True)
+    args = parser.parse_args()
+
+    init_logging_with_args(args)
+
+    # Set and create dir for download files
+    if not args.download_dir.is_dir():
+        args.download_dir.mkdir(parents=True)
+
+    # Set input file and determine if proper parameterization options are defined.
+    user_query_file = examine_parameterization(args.input_cores, args.input_accns, args.host, args.port)
+
+    ## Parse and store cores/accessions from user input query file
+    try:
+        with user_query_file.open(mode="r") as f:
+            query_list = f.read().splitlines()
+    except IOError as err:
+        logging.error(f"Unable to read user queries from inputfile '{user_query_file}' due to {err}.")
+        sys.exit()
+
+    # Set singularity cache dir from user defined path or use environment
+    if args.cache_dir and args.cache_dir.is_dir():
+        image_dl_path = Path(args.cache_dir)
+        logging.info(f"Using user-defined cache_dir: '{image_dl_path}'")
+    elif os.environ.get("NXF_SINGULARITY_CACHEDIR"):
+        image_dl_path = Path(os.environ["NXF_SINGULARITY_CACHEDIR"])
+        logging.info(
+            f"Using preferred nextflow singularity cache dir 'NXF_SINGULARITY_CACHEDIR': {image_dl_path}"
+        )
+    elif os.environ.get("SINGULARITY_CACHEDIR"):
+        image_dl_path = Path(os.environ["SINGULARITY_CACHEDIR"])
+        logging.info(
+            f"Using the default singularity installation cache dir 'SINGULARITY_CACHEDIR': {image_dl_path}"
+        )
+    else:
+        image_dl_path = Path()
+        logging.warning(f"Unable to set singularity cache dir properly, using CWD {image_dl_path}")
+
+    # Set the datasets version URL
+    if args.datasets_version_url is None:
+        container_url = DATASETS_SINGULARITY["datasets_version_url"]
+        logging.info(f"Using default 'ncbi datasets' version '{container_url}'")
+    else:
+        container_url = args.datasets_version_url
+        logging.info(f"Using user defined 'ncbi datasets' version '{container_url}'")
+
+    ## Get accessions on cores list or use user accession list directly
+    query_accessions, query_type = resolve_query_type(
+        query_list, args.host, args.port, args.input_cores, args.input_accns
+    )
+
+    # Pull or load pre-existing 'datasets' singularity container image.
+    datasets_image = Client.pull(container_url, stream=False, pull_folder=image_dl_path, quiet=True)
```

Comment:
Consider putting the whole singularity image configuration in a separate function for clarity e.g.
```
datasets_image = get_singularity_image(args.datasets_version_url, args.cache_dir)
```
(not including the `resolve_query_type` part)

Reply:
Also: is there a way to use the system `datasets` instead of the singularity image (assuming I have installed it in my PATH)? I.e. do I need to have singularity installed to run this script?

Reply:
Sorry no not at the momen. Plan is to get version 1.0 up and running via SIF container. But we shall add this functionality in a later update.

Reply:
Function to parse and define SIF image added.